wai-extra/Network/Wai/Middleware/Gzip.hs

```diff
@@ -103,17 +103,17 @@ gzip set app env sendResponse = app env $ \res ->
                                  in
                                     doesFileExist compressedVersion >>= \x ->
                                        if x
-                                         then (sendResponse $ ResponseFile s (fixHeaders hs) compressedVersion Nothing)
-                                         else (runAction (ResponseFile s hs file Nothing, nextAction))
+                                         then (sendResponse $ ResponseFile s (vary:fixHeaders hs) compressedVersion Nothing)
```

Comment:
I'm worried about this approach, given how many parts of the code need to be modified. What about redefining `sendResponse` to add the necessary header, e.g.:

```haskell
gzip set app env sendResponse' = ...
  where
    sendResponse = sendResponse . addVary
```